data-prepper-plugins/aws-plugin/src/main/java/org/opensearch/dataprepper/plugins/aws/AwsSecretsSupplier.java

```diff
@@ -107,17 +108,26 @@ public void refresh(String secretConfigId) {
         LOG.info("Finished retrieving latest secret in aws:secrets:{}.", secretConfigId);
     }
 
+    @SkipTestCoverageGenerated
```

Comment:
This is intended for generated code and maybe boilerplate. This code is significant and should be tested.

Reply:
The motivation to put this is due to coverage shortage on the exception here:

```
try {
                    Thread.sleep(waitTimeInMillis);
                } catch (InterruptedException ex) {
                    throw new RuntimeException(ex);
                }
```

but it might no avoidable with some logic improvement